coral/src/app/features/topics/request/components/AdvancedConfiguration.tsx

```diff
@@ -0,0 +1,291 @@
+  monaco: ReturnType<typeof useMonaco>,
+  advancedConfigurationOptions: TopicAdvancedConfigurationOptions[] | undefined
+) {
+  if (monaco && advancedConfigurationOptions !== undefined) {
+    const completionProvider = monaco.languages.registerCompletionItemProvider(
+      "json",
+      {
+        provideCompletionItems: autocompleteOptionNames(
+          advancedConfigurationOptions
+        ),
+      }
+    );
+    const hoverProvider = monaco.languages.registerHoverProvider("json", {
+      provideHover: wordMatchesOptionName(advancedConfigurationOptions),
+    });
+
+    return () => {
+      completionProvider.dispose();
+      hoverProvider.dispose();
+    };
+  }
+}
+
+function getRange(model: editor.ITextModel, position: Position) {
+  const word = model.getWordUntilPosition(position);
+  return {
+    startLineNumber: position.lineNumber,
+    endLineNumber: position.lineNumber,
+    startColumn: word.startColumn,
+    endColumn: word.endColumn,
+  };
+}
+
+const autocompleteOptionNames = (
+  options: TopicAdvancedConfigurationOptions[]
+) => {
+  return (model: editor.ITextModel, position: Position) => {
+    const range = getRange(model, position);
+    return {
+      suggestions: options
+        .filter((option) => !model.getValue().includes(`"${option}"`))
+        .map((option) => {
+          return {
+            label: `"${option.name}"`,
+            range: range,
+            kind: 9, //languages.CompletionItemKind.Property
+            insertText: `"${option.name}": "",`,
+          };
+        }),
+    };
+  };
+};
+
+const wordMatchesOptionName = (
+  options: TopicAdvancedConfigurationOptions[]
+) => {
+  return (model: editor.ITextModel, position: Position) => {
+    const wordAtPosition = model.getWordAtPosition(position);
+    if (wordAtPosition && isString(wordAtPosition.word)) {
+      const configuration = options.find((o) => o.name === wordAtPosition.word);
+      if (configuration !== undefined) {
+        return {
+          range: getRange(model, position),
+          contents: [
+            { value: `**${configuration.name}**` },
+            {
+              value:
+                configuration.documentation ??
+                "https://kafka.apache.org/documentation/#topicconfigs",
+            },
+          ],
+        };
+      }
+    }
+  };
+};
+
+function getUnknownConfigKeys(
+  value: string,
+  options: TopicAdvancedConfigurationOptions[]
+): string[] {
+  try {
+    const valueAsObject = JSON.parse(value);
+    const supportedKeys = new Set(options.map((o) => o.name));
+    return Object.keys(valueAsObject).filter(
+      (name) => !supportedKeys.has(name)
+    );
+  } catch {
+    return [];
+  }
+}
+
+function findMatchesFromModel(
+  keys: string[],
+  model: editor.IModel
+): { configName: string; matches: editor.FindMatch[] }[] {
+  return keys
+    .map((configName) => ({
+      configName,
+      matches: model.findMatches(
+        configName, // searchString
```

Comment:
Thanks for adding this kind of comments, that helps a lot in reading the code! ❤️